pkg/discovery/network/generic.go

```diff
@@ -62,13 +59,122 @@ func discoverGenericNetwork(clientSet kubernetes.Interface) (*ClusterNetwork, er
 }
 
 func findClusterIPRange(clientSet kubernetes.Interface) (string, error) {
+	clusterIPRange, err := findClusterIPRangeFromApiserver(clientSet)
+	if err != nil || clusterIPRange != "" {
+		return clusterIPRange, err
+	}
+
+	clusterIPRange, err = findClusterIPRangeFromServiceCreation(clientSet)
+	if err != nil || clusterIPRange != "" {
+		return clusterIPRange, err
+	}
+
+	return "", nil
+}
+
+func findClusterIPRangeFromApiserver(clientSet kubernetes.Interface) (string, error) {
 	return findPodCommandParameter(clientSet, "component=kube-apiserver", "--service-cluster-ip-range")
 }
 
+func findClusterIPRangeFromServiceCreation(clientSet kubernetes.Interface) (string, error) {
+	// find service cidr based on https://stackoverflow.com/questions/44190607/how-do-you-find-the-cluster-service-cidr-of-a-kubernetes-cluster
+	invalidSvcSpec := &v1.Service{
+		ObjectMeta: v1meta.ObjectMeta{
+			Name: "invalid-svc",
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: "1.1.1.1",
+			Ports: []v1.ServicePort{
+				{
+					Port: 443,
+					TargetPort: intstr.IntOrString{
+						IntVal: 443,
+					},
+				},
+			},
+		},
+	}
+
+	// decide which namespace to create the service
+	ns := os.Getenv("WATCH_NAMESPACE")
```

Comment:
This is set [here](https://github.com/submariner-io/submariner-operator/blob/d48d547be4ec4ed7c17dbcbc2e70dd89eb7d6fcc/pkg/subctl/operator/common/operatorpod/ensure.go#L72) as an environment variable of the operator pod. I think that this is used by operator-sdk's `GetWatchNamespace`.

Reply:
OK - could you add a comment explaining this? eg "_WATCH_NAMESPACE will be present if running in the operator pod_".

Another approach is to pass in the NS so we don't rely on the presence of an env var. The operator code implicitly knows its own namespace. I think this would make it clearer and although it would mean passing a param down the chain.

Reply:
Changed and squashed.